src/microbe_stage/ProcessStatistics.cs

```diff
@@ -412,6 +418,10 @@ public override bool Equals(object? obj)
         // This also checks for obj being null
         if (obj is AverageProcessStatistics statistics)
         {
+            GD.Print(owner.Equals(statistics.owner));
+            if (owner.Process.SpeedMultiplier != statistics.owner.Process.SpeedMultiplier)
+                GD.Print("Speed multiplier mismatch!");
```

Comment:
Is this an error condition? (also the message should be a bit more specific because if we get logs with this message in them, that will really confuse whoever is trying to determine the problem).